tests/integration/utils/comparisons.py

```diff
@@ -0,0 +1,49 @@
+# Copyright 2022 NVIDIA Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+from itertools import islice
+from typing import Any, Union
+
+import numpy as np
+
+
+def allclose(
+    a: Any,  # numpy or cunumeric array-like
+    b: Any,  # numpy or cunumeric array-like
+    rtol: float = 1e-5,
+    atol: float = 1e-8,
+    equal_nan: bool = False,
+    *,
+    diff_limit: Union[int, None] = 5,  # None means no limit at all
+) -> bool:
+    # simplify handling of scalar values
+    a, b = np.atleast_1d(a), np.atleast_1d(b)
+
+    close = np.isclose(a, b, rtol=rtol, atol=atol, equal_nan=equal_nan)
+
+    all_close = np.all(close)
+
+    if (diff_limit > 0 or diff_limit is None) and not all_close:
```

Comment:
Shouldn't you test for `diff_limit is None` first? E.g. consider:

```
>>> None > 5
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: '>' not supported between instances of 'NoneType' and 'int'
```

Reply:
Good catch! I was just thinking this helper function really itself ought to have a test..

Reply:
```suggestion
    if (diff_limit is None or diff_limit > 0) and not all_close:
```

Reply:
https://github.com/nv-legate/cunumeric/pull/457/commits/ca55c58bb4eb252acc350eccb8abd93e553ce9bf